src/components/atoms/File.tsx

```diff
@@ -1,30 +1,48 @@
-import React, { ReactElement } from 'react'
+import React, { useEffect, useState, ReactElement } from 'react'
 import { File as FileMetadata } from '@oceanprotocol/lib'
 import filesize from 'filesize'
 import classNames from 'classnames/bind'
 import cleanupContentType from '../../utils/cleanupContentType'
 import styles from './File.module.css'
+import Loader from '../atoms/Loader'
 
 const cx = classNames.bind(styles)
 
+function LoaderArea() {
+  return (
+    <div className={styles.loaderWrap}>
+      <Loader />
+    </div>
+  )
+}
+
 export default function File({
   file,
   className,
-  small
+  small,
+  isLoading
 }: {
   file: FileMetadata
   className?: string
   small?: boolean
+  isLoading?: boolean
 }): ReactElement {
+  const [loading, setLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    isLoading && setLoading(true)
+  }, [isLoading])
+
   if (!file) return null
+  console.log(isLoading, loading)
 
   const styleClasses = cx({
     file: true,
     small: small,
     [className]: className
   })
 
-  return (
+  return !loading && (isLoading === undefined || isLoading === false) ? (
```

Comment:
the file icon should be rendered at all times so there will be no layout shift. Only its content should load and be replaced